### PR TITLE
gh-122392: Fix the wrong size of the Path Browser vertical spacing for IDLE (#122392)

### PR DESCRIPTION
**Pull request by Xiaokang2022**
The Path Browser vertical interval size was originally set to `20`, but this caused display issues (such as overlapping) on some devices. Actually, we should calculate the size, and giving a fixed value will not solve the problem in all cases.

But no matter what the value is, the elements in the first row will not use it, so we can get the size of this value through the elements in the first row after the first row is drawn, and assign the value to the corresponding variable for the subsequent elements to use.

After testing, this solution worked very well on my device (Windows10 22H2 1920x1080).

* Issue: gh-122392

### CODE REVIEW

File: Lib/idlelib/tree.py
Comment (terryjreedy) [inline Lib/idlelib/tree.py]: There is a new instance for every displayed line, so this results in reduntantly recalculating dy for every line.  The time is not really a problem, but it is misleading.  Make dy a class variable.  Still read as 'self.dy', but set as 'Tree.dy'.

File: Lib/idlelib/tree.py
Comment (terryjreedy) [inline Lib/idlelib/tree.py]: Perhaps this should be `y + self.dy - 3 to retain relationship of 20 and 17.  Why I need to know when this is triggered.

File: Lib/idlelib/tree.py
Comment (terryjreedy) [inline Lib/idlelib/tree.py]: This results in too much space.  Adding ` - 3` works for me.  Also, since no complaints about 20 being too big, lets take `max(20, <expression above with subtraction>)`.

File: Misc/NEWS.d/next/IDLE/2024-10-04-15-34-34.gh-issue-122392.V8K3w2.rst
Comment (terryjreedy) [inline Misc/NEWS.d/next/IDLE/2024-10-04-15-34-34.gh-issue-122392.V8K3w2.rst]:
```suggestion
Increase currently inadequate vertical spacing for the IDLE browsers (path, module, and stack) on high-resolution monitors.
```

**Review by terryjreedy — CHANGES_REQUESTED** [review]
See additional new comment on issue.

**Comment by Xiaokang2022** [comment]
I have made the requested changes; please review again.

**Review by terryjreedy — APPROVED** [review]

**Comment by terryjreedy** [comment]
@serhiy-storchaka, @ned-deily, @ronaldoussoren: I am pretty sure that this should work on linux/macOS, but would any of you like to test before I merge?  Open IDLE, select File => pathbrowser, open /Lib, the underscores and descenders on `pyg` should be complete (they likely would not have  been before if you have a high-def screen).  If normal screen, the space between folder icons should be about the same size as the icons.

**Comment by Xiaokang2022** [comment]
I tested it on Arch Linux and it looks like this:

**Comment by Xiaokang2022** [comment]
Hello, @terryjreedy

Regarding your [second comment](https://github.com/python/cpython/pull/124975#discussion_r1788118301) from the above review, I think there may be something wrong with some. After I looked through the source code in the past, I found that the reason for [the value](https://github.com/python/cpython/blame/b3e2c0291595edddc968680689bec7707d27d2d1/Lib/idlelib/tree.py#L213) of `17` here may be caused by you missing it in a change from 10 years ago. (https://github.com/python/cpython/commit/9a6f8e18662c67cd41bf54ef3b8dbf2b7f5a3aeb)

c5305b8ae15aa0b1c15fb4d59a1781f8

So I think my previous code might be correct. Of course, there may be some problems that I have not taken into account, and if so, please forgive me and point them out.

File: Lib/idlelib/tree.py
Comment (terryjreedy) [inline Lib/idlelib/tree.py]: Good catch
```suggestion
                return y + TreeNode.dy
```

**Comment by terryjreedy** [comment]
Good digging.  I could think that if 17 has not resulted in a report, maybe it was OK.  But people mostly do not report glitches. Better safe than sorry.

**Comment by terryjreedy** [comment]
I intend to merge this after 3.13.0 is released, which is scheduled to be within a day.

**Pull request merged**
